eng/tools/GenerateFiles/Directory.Build.targets.in

```diff
@@ -54,10 +54,6 @@
           Condition=" '$(IsServicingBuild)' != 'true' ">${MicrosoftAspNetCoreAppRefVersion}</TargetingPackVersion>
     </KnownFrameworkReference>
 
-    <PackageReference Include="Microsoft.Net.Compilers.Toolset"
-      Version="${MicrosoftNetCompilersToolsetVersion}"
-      PrivateAssets="all"
-      IsImplicitlyDefined="true" />
```

Comment:
Do we know why this was here before?

Reply:
I don't know, we'll see if something breaks. I spoke to Doug and he's going to follow up and do some research.

Reply:
@wtgodbe introduced this long ago (for a 3.0.0 preview) in 23a6e3ead9a / #11818. We were reacting to a Roslyn breaking change and needed a newer toolset than Arcade (think we set `$(UsingToolMicrosoftNetCompilers)` at one point) or `msbuild` references by default. Maestro++ managed the version for a while but we stopped that at some point. @BrennanConroy and @pranavkm were among the few to change the version after we made the version manual.

I don't remember the details of the breaking change but it had something to do w/ nullability annotations and @roji did relevant work in the efcore repo. @jaredpar may know though he wasn't involved in the conversation back in the summer of 2019.

----

As a follow up, we should do one of

1. change the manually-controlled toolset version and re-introduce this package reference
2. clean up https://github.com/dotnet/aspnetcore/blob/8ee0b03d15fe5a3fbba86fbcf165bd8e5da53d6e/eng/Versions.props#L160
3. enable  `$(UsingToolMicrosoftNetCompilers)`, meaning Arcade updates will bump the toolset version on occasion

I don't know enough about the toolset to have an opinion on which choice is best. @jaredpar❔

Reply:
In general repositories should avoid explicitly referencing this package and instead prefer the one that just comes with arcade. The arcade version is updated at a regular cadence (roughly every other week). The idea being that this update cadence means we're regularly pushing new versions of the compiler to our partners in .NET. 

This type of change where the version is pinned should be reserved for cases where a breaking change, bug, etc ... hits the repo and you need to pause the updates until we get a fix out. That appears to be what happened here except we forget to undo the pin once the bug was fixed.

Reply:
>  The arcade version is updated at a regular cadence (roughly every other week).

Right now, the Arcade version (see https://github.com/dotnet/arcade/blob/05634736a53ee65b0d19cf913d77d1f5d4293ebc/eng/Versions.props#L37) is far behind what the SDK uses by default (see https://github.com/dotnet/sdk/blob/ef6249b3b034b12f127b0d3a72ea428979f44413/eng/Versions.props#L120). Right now Arcade is only at `v3.10.0-4.21329.37` while the SDK is at `v4.0.0-2.21354.7`. I suggest we simply remove our confusing `$(MicrosoftNetCompilersToolsetVersion)` property and leave `$(UsingToolMicrosoftNetCompilers)` unset i.e. my option (2). Agreed @dotnet/aspnet-build @jaredpar❔